### ISSUE TITLE
New incompatibility between nicematrix and bidi

### ISSUE THREAD

**Issue by logological** [kvafa/bidi#42]
## Brief outline of the issue

With a recent (2024-06-01-ish) release of LaTeX, use of the `bidi` package seems to break `nicematrix` >= 6.28.  According to the discussion at Issue latex3/latex2e#1368, this may be because the `array` package, whose definitions `bidi` redefines, was recently changed.

I also reported this issue by e-mail to the `nicematrix` maintainer.

## Check/indicate
- [ ] Relevant to the `xepersian` package
- [x] [The `bidi` package issue tracker](https://github.com/kvafa/bidi/issues) has been searched for similar issues?
- [x] [The `xepersian` package issue tracker](https://github.com/kvafa/xepersian/issues) has been searched for similar issues?
- [ ] Links to <tex.stackexchange.com> discussion if appropriate
- [ ] Links to <qa.parsilatex.com> discussion if appropriate

## Minimal example showing the issue

```tex
\documentclass{article}

\usepackage{nicematrix}
\usepackage{bidi}

\begin{document}
\begin{NiceTabular}{lllll}
a & b & c & d & e \\
\end{NiceTabular}
\end{document}
```

## Expected behavior

The minimal example should compile, but instead produces the following error message:

```
(./test.aux) (/opt/texlive/2024/texmf-dist/tex/latex/base/ts1cmr.fd)
! Extra alignment tab has been changed to \cr.
<template> \endtemplate

l.9 \end{NiceTabular}
```

## Log and PDF files  

[test.log](https://github.com/user-attachments/files/15744843/test.log)

**Comment by logological** [comment]
With the latest release of `array` (2.6d on 2024-06-14) the problem is no longer reproducible.  However, the discussion in latex3/latex2e#1368 suggests that the root of the problem is `bidi`'s redefinition of internal things from `array`, and so it might be better if `bidi` were eventually updated to play nice with the new definitions.

**Comment by seloumi** [comment]
Same issue with tabularray
https://tex.stackexchange.com/q/723112/54817

**Comment by ghost** [comment]
I have fixed this in version `40.1` of the `bidi` package and uploaded it about a week ago. It should already be available on TeXLive via update.

If this addresses your issue, then please consider closing this issue.

Please also note that this repository is unmaintained. The `bidi` package has a [new repository](https://github.com/xepersian/bidi).

**Comment by logological** [comment]
Confirmed issue is no longer reproducible with bidi v40.4 (Release John Willard Milnor), 2024/12/15 and array 2024-06-01.  (I didn't test with any older version of array.)
